pkg/protocol/protobuf/common.proto

```diff
@@ -29,3 +29,28 @@ message OffPeakChargingTimes {
         Void weekdays = 2;
     }
 }
+
+message ChargeSchedule {
+    uint64                      id                      = 1; // datetime in epoch time
+    string                      name                    = 2;
+    int32                       days_of_week            = 3;
+    bool                        start_enabled           = 4;
+    int32                       start_time              = 5; // 24h in minutes
+    bool                        end_enabled             = 6;
+    int32                       end_time                = 7; // 24h in minutes
+    bool                        one_time                = 8;
+    bool                        enabled                 = 9;
+    float                       latitude                = 10;
+    float                       longitude               = 11;
+}
+
+message PreconditionSchedule {
```

Comment:
nit: can we support generating protobufs in makefile?
https://github.com/teslamotors/vehicle-command/blob/main/Makefile
we do something like this in fleet telemetry as well

Reply:
See above. created https://github.com/teslamotors/vehicle-command/issues/307 to improve our proto generation.